inc/appliance.class.php

```diff
@@ -0,0 +1,436 @@
+      echo "<td>";
+      Manufacturer::dropdown(['value' => $this->fields["manufacturers_id"], 'rand' => $randDropdown]);
+      echo "</td></tr>\n";
+
+      echo "<tr class='tab_bg_1'>";
+      $randDropdown = mt_rand();
+      echo "<td><label for='dropdown_groups_id_tech$randDropdown'>".__('Group in charge of the appliance')."</label></td>";
+      echo "<td>";
+      Group::dropdown([
+         'name'      => 'groups_id_tech',
+         'value'     => $this->fields['groups_id_tech'],
+         'entity'    => $this->fields['entities_id'],
+         'condition' => ['is_assign' => 1],
+         'rand' => $randDropdown
+      ]);
+
+      echo "</td>";
+      echo "<td><label for='dropdown_applianceenvironments_id$randDropdown'>".ApplianceEnvironment::getTypeName(1)."</label></td>";
+      echo "<td>";
+      ApplianceEnvironment::dropdown(['value' => $this->fields["applianceenvironments_id"], 'rand' => $randDropdown]);
+      echo "</td></tr>\n";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td><label for='textfield_serial$rand'>".__('Serial number')."</label></td>";
+      echo "<td >";
+      Html::autocompletionTextField($this, 'serial', ['rand' => $rand]);
+      echo "</td>";
+
+      echo "<td><label for='textfield_otherserial$rand'>".sprintf(__('%1$s%2$s'), __('Inventory number'), $tplmark).
+           "</label></td>";
+      echo "<td>";
+
+      $objectName = autoName($this->fields["otherserial"], "otherserial",
+                             (isset($options['withtemplate']) && ($options['withtemplate'] == 2)),
+                             $this->getType(), $this->fields["entities_id"]);
+      Html::autocompletionTextField(
+         $this,
+         'otherserial',
+         [
+            'value'     => $objectName,
+            'rand'      => $rand
+         ]
+      );
+
+      echo "</td></tr>\n";
+
+      echo "<tr class='tab_bg_1'>";
+      $randDropdown = mt_rand();
+      echo "<td><label for='dropdown_users_id$randDropdown'>".__('User')."</label></td>";
+      echo "<td>";
+      User::dropdown(['value'  => $this->fields["users_id"],
+                           'entity' => $this->fields["entities_id"],
+                           'right'  => 'all',
+                           'rand'   => $randDropdown]);
+      echo "</td>";
+
+      $randDropdown = mt_rand();
+      echo "<td><label for='dropdown_groups_id$randDropdown'>".__('Group')."</label></td>";
+      echo "<td>";
+      Group::dropdown([
+         'value'     => $this->fields["groups_id"],
+         'entity'    => $this->fields["entities_id"],
+         'condition' => ['is_itemgroup' => 1],
+         'rand'      => $randDropdown
+      ]);
+
+      echo "</td></tr>";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>" . __('Associable to a ticket') . "</td><td>";
+      Dropdown::showYesNo('is_helpdesk_visible', $this->fields['is_helpdesk_visible']);
+      echo "</td></tr>\n";
+
```

Comment:
Missing Item to link and Comments

Reply:
Comments has been added on form. As for adding an item, it seems this is also related to ApplianceRelation.